src/tools/rbd/action/MirrorPool.cc

```diff
@@ -259,16 +259,48 @@ void get_enable_arguments(po::options_description *positional,
 }
 
 int execute_enable_disable(const std::string &pool_name,
-                           rbd_mirror_mode_t mirror_mode) {
+                           const std::string &mode) {
   librados::Rados rados;
   librados::IoCtx io_ctx;
+  rbd_mirror_mode_t next_mirror_mode;
+  rbd_mirror_mode_t current_mirror_mode;
+
+  if (mode == "disable") {
```

Comment:
@dillaman  yeah, i will pass it in next PR. I was having same thing in my mind previously but i thought we can save passing extra argument as a string so moved logic here form execute_enable function.